src/Avalonia.Animation/AnimatorDrivenTransition.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using Avalonia.Animation.Animators;
+
+namespace Avalonia.Animation
+{
+    public class AnimatorDrivenTransition<T, TAnimator> : Transition<T> where TAnimator : Animator<T>, new()
```

Comment:
I was actually thinking about creating this class before, but decided to not add anything to the public API.
Unfortunately it cannot be internal.

Reply:
Probably it makes more sense to be abstract

Reply:
I guess it could be internal after library merge. But I was thinking that it might be useful for users writing their own transitions/animators.

Reply:
Library merge would help with class visibility in both Base and Visual assemblies (where we have different sets of transitions), but even now it can be solved with InternalVisibleTo.

The bigger problem is that C# doesn't allow to inherit internal classes, if your one is public.

Reply:
This could technically be solved by having implementation classes but that will cause indirections again. Could think about a mixin based solution if we really don't want to expose such API surface.